src/main/java/frc/robot/commandGroups/CommandGroups.java

```diff
@@ -0,0 +1,93 @@
+package frc.robot.commandGroups;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.GripperState;
+import frc.robot.scoreStates.ScoreState;
+import frc.robot.subsystems.conveyor.Conveyor;
+import frc.robot.subsystems.elevator.Elevator;
+import frc.robot.subsystems.elevator.ElevatorConstants;
+import frc.robot.subsystems.gripper.Gripper;
+import frc.robot.subsystems.gripper.GripperConstants;
+import frc.robot.subsystems.intake.Intake;
+import frc.robot.subsystems.shooter.Shooter;
+import frc.robot.subsystems.shooter.ShooterConstants;
+import java.util.function.Supplier;
+
+public class CommandGroups {
+    private static CommandGroups INSTANCE;
+    private final Intake intake;
+    private final Gripper gripper;
+    private final Elevator elevator;
+    private final Shooter shooter;
+    private final Conveyor conveyor;
+    private final GripperState gripperState;
+
+    private CommandGroups() {
+        intake = Intake.getInstance();
+        gripper = Gripper.getInstance();
+        elevator = Elevator.getInstance();
+        shooter = Shooter.getInstance();
+        conveyor = Conveyor.getInstance();
+        gripperState = new GripperState();
+    }
+
+    public static CommandGroups getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new CommandGroups();
+        }
+        return INSTANCE;
+    }
+
+    public Command elevatorGripperMinPosition() {
```

Comment:
1. "Min" is a very general term, it's not clear what you mean in that. Change everywhere to something more specific.
2. Command names should start with a verb.

Reply:
d7ae473